components/tidb_query/src/codec/mysql/decimal.rs

```diff
@@ -1793,20 +1794,13 @@ impl ConvertTo<Decimal> for Real {
 }
 
 impl ConvertTo<Decimal> for &[u8] {
+    // FIXME, the err handle is not exactly same as TiDB's,
+    //  TiDB's seems has bug, fix this after fix TiDB's
     #[inline]
     fn convert(&self, ctx: &mut EvalContext) -> Result<Decimal> {
-        let dec = match Decimal::from_bytes(self)? {
-            Res::Ok(d) => d,
-            Res::Overflow(d) => {
-                ctx.handle_overflow_err(Error::overflow("DECIMAL", ""))?;
-                d
-            }
-            Res::Truncated(d) => {
-                ctx.handle_truncate(true)?;
-                d
-            }
-        };
-        Ok(dec)
+        let r = Decimal::from_bytes(self)?;
```

Comment:
One-row task split in three steps XD

Reply:
without split, after rustfmt, it will look ungly
![image](https://user-images.githubusercontent.com/21231345/64752004-a55dd200-d550-11e9-95ad-2187beded62d.png)